### PR TITLE
[optimizer][docs] Add Tutorial for Optimizer API

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Add Tutorial for Optimizer API

### CODE REVIEW

File: docs/optimizer/optimize.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+The ONNXScript `Optimizer` tool provides the user with the functionality to optimize an ONNX model by performing optimizations and clean-ups such as constant folding, dead code elimination, etc.
+
+## Usage
+
+In order to utilize the optimizer tool,
+
+```python
+    import onnxscript
+    onnxscript.optimizer.optimize(model)
```
Comment (justinchuby) [inline docs/optimizer/optimize.md:13]:
```suggestion
import onnxscript

onnxscript.optimizer.optimize(model)
```

File: docs/optimizer/optimize.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Optimizing a Model using the Optimizer
+
+## Introduction
+
+The ONNXScript `Optimizer` tool provides the user with the functionality to optimize an ONNX model by performing optimizations and clean-ups such as constant folding, dead code elimination, etc.
```
Comment (justinchuby) [inline docs/optimizer/optimize.md:5]:
```suggestion
The ONNX Script `Optimizer` tool provides the user with the functionality to optimize an ONNX model by performing optimizations and clean-ups such as constant folding, dead code elimination, etc.
```

File: docs/index.md [hunk 3]
```diff
@@ -100,6 +100,7 @@ tutorial/index
 api/index
 intermediate_representation/index
 auto_examples/index
+optimizer/index
```
Comment (justinchuby) [inline docs/index.md:103]:
```suggestion
optimizer/index
auto_examples/index
```

We can move it up

File: docs/optimizer/optimize.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## Description of optimizations applied by `onnxscript.optimizer.optimize`
+
+```{eval-rst}
+.. autosummary::
+    :nosignatures:
+
+    onnxscript.optimizer.constant_folding.fold_constants
+    onnxscript.optimizer.copy_propagation.do_copy_propagation
+    onnxscript.optimizer.copy_propagation.do_sequence_simplification
```
Comment (gramalingam) [inline docs/optimizer/optimize.md:33]: I think both of above (copy propagation and sequence simplification) have been directly incorporated into the constant-folding optimizer. I suspect that we may be able to eliminate these older standalone optimization components (in terms of code and files ... but those optimizations are being done, so it is okay to mention them by name, but not this specific module-name.

Comment (shubhambhokare1) [reply docs/optimizer/optimize.md]: I went ahead and eliminate the redundant standalone optimization components. Converted the table from autogeneration to manual mode for the time being to make it easier to highlight the passes happening inside constant_folding.

File: docs/rewriter/rewrite_patterns.md [hunk 5]
```diff
@@ -28,10 +28,10 @@ We will show how we can find a subgraph matching this computation and replace it
 Firstly, include all the rewriter relevant imports.
 
 ```python
-    from onnxscript.rewriter import pattern
-    from onnxscript import ir
+from onnxscript.rewriter import pattern
+from onnxscript import ir
 
-    _op = pattern.onnxop
+_op = pattern.onnxop
```
Comment (justinchuby) [inline docs/rewriter/rewrite_patterns.md:34]: I would check the code blocks below too. There are some more blocks that can be dedented

**Review by justinchuby — APPROVED** [review]

**Pull request merged**